snudown.h

```diff
@@ -0,0 +1,118 @@
+/* snudown.h - header for snudown */
```

Comment:
`#include` is dumb and copy pastes code. If you put implementations in headers, they get copy pasted into every module that includes it which means redefinitions.